src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_debug.c

```diff
@@ -1,25 +1,1002 @@
+                            }
+                        } else if (CHECK_BTN_ALL(input->press.button, BTN_CLEFT)) {
+                            if (INV_CONTENT(ITEM_MOON_TEAR) == ITEM_NONE) {
+                                gSaveContext.save.inventory.items[slot] = ITEM_DEED_OCEAN;
+                            } else if ((INV_CONTENT(ITEM_MOON_TEAR) >= ITEM_DEED_LAND) &&
+                                       (INV_CONTENT(ITEM_MOON_TEAR) <= ITEM_DEED_OCEAN)) {
+                                gSaveContext.save.inventory.items[slot] = INV_CONTENT(ITEM_MOON_TEAR) - 1;
+                            }
+                        }
+                    } else if (slot == SLOT_TRADE_KEY_MAMA) {
+                        if (CHECK_BTN_ALL(input->press.button, BTN_CRIGHT)) {
+                            if (INV_CONTENT(ITEM_ROOM_KEY) == ITEM_NONE) {
+                                gSaveContext.save.inventory.items[slot] = ITEM_ROOM_KEY;
+                            } else if ((INV_CONTENT(ITEM_ROOM_KEY) >= ITEM_ROOM_KEY) &&
+                                       (INV_CONTENT(ITEM_ROOM_KEY) <= ITEM_ROOM_KEY)) {
+                                gSaveContext.save.inventory.items[slot] = INV_CONTENT(ITEM_ROOM_KEY) + 1;
+                            }
+                        } else if (CHECK_BTN_ALL(input->press.button, BTN_CLEFT)) {
+                            if (INV_CONTENT(ITEM_ROOM_KEY) == ITEM_NONE) {
+                                gSaveContext.save.inventory.items[slot] = ITEM_LETTER_MAMA;
+                            } else if ((INV_CONTENT(ITEM_ROOM_KEY) >= ITEM_LETTER_MAMA) &&
+                                       (INV_CONTENT(ITEM_ROOM_KEY) <= ITEM_LETTER_MAMA)) {
+                                gSaveContext.save.inventory.items[slot] = INV_CONTENT(ITEM_ROOM_KEY) - 1;
+                            }
+                        }
+                    } else if (CHECK_BTN_ALL(input->press.button, BTN_CRIGHT)) {
+                        if (INV_CONTENT(ITEM_LETTER_TO_KAFEI) == ITEM_NONE) {
+                            gSaveContext.save.inventory.items[slot] = ITEM_LETTER_TO_KAFEI;
+                        } else if ((INV_CONTENT(ITEM_LETTER_TO_KAFEI) >= ITEM_LETTER_TO_KAFEI) &&
+                                   (INV_CONTENT(ITEM_LETTER_TO_KAFEI) <= ITEM_LETTER_TO_KAFEI)) {
+                            gSaveContext.save.inventory.items[slot] = INV_CONTENT(ITEM_LETTER_TO_KAFEI) + 1;
+                        }
+                    } else if (CHECK_BTN_ALL(input->press.button, BTN_CLEFT)) {
+                        if (INV_CONTENT(ITEM_LETTER_TO_KAFEI) == ITEM_NONE) {
+                            gSaveContext.save.inventory.items[slot] = ITEM_PENDANT_MEMORIES;
+                        } else if ((INV_CONTENT(ITEM_LETTER_TO_KAFEI) >= ITEM_PENDANT_MEMORIES) &&
+                                   (INV_CONTENT(ITEM_LETTER_TO_KAFEI) <= ITEM_PENDANT_MEMORIES)) {
```

Comment:
huh

Reply:
Just in case this wasn't clear at first since this can look kind of funny:
`INV_CONTENT(ITEM_LETTER_TO_KAFEI)` means "look at the slot that belongs to `ITEM_LETTER_TO_KAFEI`, and read the item in that slot".
So because `ITEM_LETTER_TO_KAFEI` and `ITEM_PENDANT_MEMORIES` share the same slot, it's using `ITEM_LETTER_TO_KAFEI` to get the slot, but then is checking for `ITEM_PENDANT_MEMORIES` (with the same `==` equivalent comment as above)